src/lava/lib/optimization/solvers/generic/solver.py

```diff
@@ -201,6 +253,56 @@ def _get_requirements_and_protocol(self,
         else:
             raise NotImplementedError(str(backend) + backend_msg)
 
+    def _get_run_config(self, backend, measure_time, measure_power, timeout):
+        do_benchmark = measure_time or measure_power
+        if backend in CPUS:
+            pdict = {self.solver_process: self.solver_model,
+                     ReadGate: ReadGatePyModel,
+                     Dense: PyDenseModelFloat,
+                     StochasticIntegrateAndFire: StochasticIntegrateAndFireModel
+                     }
+            run_cfg = Loihi1SimCfg(exception_proc_model_map=pdict,
+                                   select_sub_proc_model=True)
+        elif backend in NEUROCORES:
+            pdict = {self.solver_process: self.solver_model,
+                     StochasticIntegrateAndFire:
+                         StochasticIntegrateAndFireModelSCIF,
+                     }
+            pre_run_fxs, post_run_fxs = [], []
+            if do_benchmark:
+                if timeout == -1:
+                    raise ValueError("For energy or time measurements timeout "
+                                     "cannot be -1")
+                if measure_power:
+                    pre_run_fxs, post_run_fxs = \
+                        self._benchmarker.get_power_measurement_cfg(
+                            num_steps=timeout + 1)
+                elif measure_time:
+                    pre_run_fxs, post_run_fxs = \
+                        self._benchmarker.get_time_measurement_cfg(
+                            num_steps=timeout + 1)
+
+            run_cfg = Loihi2HwCfg(exception_proc_model_map=pdict,
+                                  select_sub_proc_model=True,
+                                  pre_run_fxs=pre_run_fxs,
+                                  post_run_fxs=post_run_fxs
+                                  )
+        else:
+            raise NotImplementedError(str(backend) + backend_msg)
+        return run_cfg
+
+    def _validated_cost(self, target_cost):
+        if target_cost % int(target_cost) != 0:

```

Comment:
This will throw an error for target_cost == 0 (which is actually the default).  Would suggest to replace it with
`if target_cost !=0 and target_cost % int(target_cost) != 0:`

Note: I also mentioned this in another PR that contained this code.